mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -169,6 +171,17 @@ public DaoDeleteMethodGenerator(
         "$T boundStatementBuilder = $L.boundStatementBuilder()",
         BoundStatementBuilder.class,
         statementName);
+    if (attributes != null) {
+      deleteBuilder.addStatement(
+          "boundStatementBuilder = populateBoundStatementWithAttributes(boundStatementBuilder, $S, $S, $S, $L, $L, $L, $S)",
+          attributes.ExecutionProfileName(),
+          attributes.consistencyLevel(),
+          attributes.serialConsistencyLevel(),
+          attributes.idempotence(),
+          attributes.pageSize(),
+          attributes.timeout(),
+          attributes.routingKeyspace());
+    }
     if (statementAttributeParam != null) {
```

Comment:
This allows both the annotation and the parameter.
1. :+1: I think this is the right thing to do. I can see cases where you might want one attribute static and another dynamic.
2. We should cover that in tests.
3. There is going to be a problem with the way default values are handled. Currently the parameter version of `populateBoundStatementWithAttributes` always sets the values, it might overwrite an annotation value with a default that the user didn't set explicitly. Actually I think we might in fact need a way to tell which values were set on `StatementAttributes` (sorry I made you change that in the other PR): either `xxxWasSet` methods, or we make every getter return an optional.